BitwardenShared/UI/Platform/Application/Views/BitwardenTextField.swift

```diff
@@ -81,19 +82,31 @@ struct BitwardenTextField<TrailingContent: View>: View {
     /// The main content for the view, containing the title label and text field.
     @ViewBuilder private var contentView: some View {
         HStack(spacing: 8) {
-            VStack(alignment: .leading, spacing: 2) {
-                // This preserves space for the title to lay out above the text field when
-                // it transitions from the centered to top position. But it's always hidden and
-                // the text above is the one that moves during the transition.
-                titleText(showPlaceholder: false)
-                    .hidden()
-
-                textField
-            }
-            .overlay(alignment: showPlaceholder ? .leading : .topLeading) {
+            ZStack(alignment: showPlaceholder ? .leading : .topLeading) {
                 // The placeholder and title text which is vertically centered in the view when the
                 // text field doesn't have focus and is empty and otherwise displays above the text field.
                 titleText(showPlaceholder: showPlaceholder)
+
+                // Since the title changes font size based on if it's the placeholder, this hidden
+                // view preserves space to show the title in it's placeholder form. This prevents
+                // the field from changing size when the placeholder's visibility changes.
+                titleText(showPlaceholder: true)
+                    .hidden()
+
+                VStack(alignment: .leading, spacing: 2) {
+                    // This preserves space for the title to lay out above the text field when
+                    // it transitions from the centered to top position. But it's always hidden and
+                    // the text above is the one that moves during the transition.
+                    titleText(showPlaceholder: false)
+                        .hidden()
```

Comment:
This is possible, but it's kind of quirky in practice. The size calculations don't always adjust immediately when changing the font size. And it ends up breaking snapshot tests, because a second layout pass is needed to calculate the height of the text. I don't think it's worth it, unless we know that these fields are causing memory issues.